test/libsolidity/SolidityNameAndTypeResolution.cpp

```diff
@@ -237,8 +237,8 @@ BOOST_AUTO_TEST_CASE(double_function_declaration)
 {
 	char const* text = R"(
 		contract test {
-			function fun() { uint x; }
-			function fun() { uint x; }
+			function fun() { uint x; x; }
```

Comment:
This will actually trigger another warning (in a different PR): Side-effect free statement.
I think you can silence both by using `x = 1`.